pkg/vpcmodel/explainabilityConnectivity.go

```diff
@@ -476,3 +482,59 @@ func (v *VPCConnectivity) getConnection(c *VPCConfig, src, dst Node) (conn *deta
 	}
 	return conn, nil
 }
+
+func (details *rulesAndConnDetails) updateRespondRules(c *VPCConfig, connQuery *connection.Set) error {
+	for _, srcDstDetails := range *details {
+		// respond rules are relevant if connection has a TCP component and non-stateful filter (NACL at the moment)
+		// are relevant for <src, dst>
+		if !respondRulesRelevant(srcDstDetails.conn, srcDstDetails.filtersRelevant) {
+			continue
+		}
+		connForResp := newTCPSet()
+		if connQuery != nil {
+			connForResp = connForResp.Intersect(connQuery)
+		}
+		respondRules, err := c.getRespondRules(srcDstDetails.src, srcDstDetails.dst, connForResp)
+		if err != nil {
+			return err
+		}
+		srcDstDetails.respondRules = respondRules
+	}
+	return nil
+}
+
+func respondRulesRelevant(conn *detailedConn, filtersRelevant map[string]bool) bool {
+	return !conn.tcpRspEnable.Union(conn.tcpRspDisable).IsEmpty() && filtersRelevant[NaclLayer]
+}
+
+// gets the NACL rules that enables/disables respond for connection conn, assuming nacl is applied
+func (c *VPCConfig) getRespondRules(src, dst Node,
+	conn *connection.Set) (respondRules *rulesConnection, err error) {
+	ingressAllowPerLayer, egressAllowPerLayer := rulesInLayers{}, rulesInLayers{}
+	ingressDenyPerLayer, egressDenyPerLayer := rulesInLayers{}, rulesInLayers{}
```

Comment:
defines these for variables inside the ifs

Reply:
I can't since their value is cross loop